Added importing and exporting for new structure

ZhangTerrence opened the pull request:
Added:
- Importing csv files
- Exporting csv and pdf files

ZhangTerrence commented:
The new `GET /exchanges` has a request query parameter `userId` which will query the `exchanges` by `userId`. If `userId` is null, then all exchanges are returned.

miguel-merlin commented on src/main/java/com/sarapis/orservice/controller/ExchangeController.java:
```diff
@@ -0,0 +1,29 @@
+    private final ExchangeService exchangeService;
+
+    @GetMapping()
+    public ResponseEntity<List<ExchangeDTO.Response>> getAllExchanges(
```
Add pagination instead of returning the whole list

miguel-merlin commented on src/main/java/com/sarapis/orservice/controller/ExchangeController.java:
```diff
@@ -0,0 +1,29 @@
+@RequestMapping("/exchanges")
+@RequiredArgsConstructor
+@Slf4j
+public class ExchangeController {
```
`nit` Change this to DataExachangeController?

miguel-merlin commented on src/main/java/com/sarapis/orservice/controller/ExportController.java:
```diff
@@ -0,0 +1,54 @@
+
+    @PostMapping(produces = "application/zip")
+    public void exportFiles(HttpServletResponse response, @RequestBody ExchangeDTO.Request request) {
+        response.setStatus(HttpServletResponse.SC_OK);
```
Add all the `response.set{}` in a private helper function for better readability

miguel-merlin commented on src/main/java/com/sarapis/orservice/controller/ExportController.java:
```diff
@@ -0,0 +1,54 @@
+        response.setHeader("Content-Disposition", "attachment; filename=\"orservice.zip\"");
+        response.setContentType("application/zip");
+
+        try {
```
Is there anyway you can move this logic to the service layer?

miguel-merlin commented on src/main/java/com/sarapis/orservice/controller/ExportController.java:
```diff
@@ -0,0 +1,54 @@
+
+            switch (request.getFormat()) {
+                case CSV:
+                    bytes += organizationService.writeCsv(zipOutputStream);
```
Now that we have all the entities modeled we need to add them

miguel-merlin commented on src/main/java/com/sarapis/orservice/controller/ImportController.java:
```diff
@@ -0,0 +1,71 @@
+            @RequestPart("file") MultipartFile file,
+            @CookieValue(value = "updatedBy", required = false, defaultValue = "SYSTEM") String updatedBy
+    ) {
+        try {
```
Some comment here. Can you move this logic to the service layer?

miguel-merlin commented on src/main/java/com/sarapis/orservice/controller/ImportController.java:
```diff
@@ -0,0 +1,71 @@
+
+            switch (format) {
+                case CSV: {
+                    if (!("text/csv".equals(file.getContentType()))) {
```
set `text/csv` to a `private static String`

miguel-merlin commented on src/main/java/com/sarapis/orservice/controller/ImportController.java:
```diff
@@ -0,0 +1,71 @@
+                    }
+
+                    switch (Objects.requireNonNull(file.getOriginalFilename())) {
+                        case "organizations.csv":
```
Since you will be receiving multiple files `{organizations.csv, services.csv}` create a HashMap with the following structure
```
"organizations.csv":  functionToHandleOrganization
```
Then in the for loop just do `hashmap[file_name]` make sure to handle exceptions correclty

ZhangTerrence commented:
- Addressed the comments, which required some major refactoring
- Renamed `Exchange` entity to `DataExchange`
- Moved `timestamp` attribute from `FileImport` entity to `DataExchange` entity
- `POST /import` and `POST /export` now should support multiple files. Still need to add the necessary methods for the other entities though so currently, only `Organization` can be imported and exported

miguel-merlin commented:
@ZhangTerrence Please solve the merge conflicts and I'll merge your PR
